README.md

```diff
@@ -147,23 +144,26 @@ functions** as
 [WebAssembly imports](https://webassembly.github.io/spec/core/syntax/modules.html#imports)
 (all of them defined in the `oak` module):
 
--   `channel_read: (i64, i32, i32) -> i32`: Reads data from the specified
-    channel.
+-   `channel_read: (i64, i32, i32) -> i32`: Reads a single message from the
```

Comment:
How about we use an "out" pointer parameter to output the number of actual bytes, and let the call return a status code instead. Something like this will be necessary anyways when we need to read and write handles in addition to bytes, so we may as well do it now.
Basically this starts to look more and more like https://fuchsia.googlesource.com/fuchsia/+/refs/heads/master/zircon/docs/syscalls/channel_write.md